### PR TITLE
Allow seeking on local files (fixes #130)

### LINKED ISSUE #130: Unable to seek file
When playing video files (I tried ```video/mp4``` files), using any of the command ```seek```, `ffwd` or `rewind` causes the chromecast device to restart the playback from scratch.
From what I managed to gather, it seems that the media server should implement `http progressive download` for the chromecast to be able to seek properly

### PR DESCRIPTION
**Pull request by skorokithakis**
This implements `Range` header support and allows seeking.

### CODE REVIEW

File: catt/http_server.py [hunk 1]
```diff
@@ -1,9 +1,45 @@
...
+
+
+def copy_byte_range(infile, outfile, start=None, stop=None, bufsize=16 * 1024):
+    """Like shutil.copyfileobj, but only copy a range of the streams.
+
+    Both start and stop are inclusive.
+    """
+    if start is not None:
+        infile.seek(start)
+    while 1:
```
Comment (theychx) [inline catt/http_server.py]: Is this preferable to ```while True```?

File: catt/http_server.py [hunk 2]
```diff
@@ -1,9 +1,45 @@
...
+
+def parse_byte_range(byte_range):
+    """Returns the two numbers in 'bytes=123-456' or throws ValueError.
+
+    The last number or both numbers may be None.
+    """
+    if byte_range.strip() == "":
+        return None, None
+
+    m = BYTE_RANGE_RE.match(byte_range)
```
Comment (theychx) [inline catt/http_server.py]: I don't like one-letter variable names (except in comprehensions).

**Comment by theychx** [comment]
I can't get this to fail. :champagne:

**Comment by theychx** [comment]
flake issues:
https://lintlyci.github.io/Flake8Rules/rules/W605.html
As for the "F841 local variable 'e' is assigned to but never used", I still struggle to understand why this is only a problem with py3.6.

**Comment by skorokithakis** [comment]
Alright, updated!

**Pull request merged**